packages/remote-viewport/src/remote-machine.ts

```diff
@@ -1,11 +1,22 @@
+import { ReadonlyMat4, mat4 } from 'gl-matrix';
+import { assign, createMachine } from 'xstate';
+
 import { Viewport } from '@itk-viewer/viewer/viewport.js';
-import { ActorRef, assign, createMachine } from 'xstate';
+
+type RendererProps = {
+  density: number;
+  cameraPose: ReadonlyMat4;
+};
+
+type RendererPatch = Partial<RendererProps>;
```

Comment:
In order to:

1. Keep the logic of how this is handled on the receiving end simple and clear.
2. Keep the impact of the effect of the order of events consistently applied.

the patch should be an Array of Event's that can be sent to the renderer.

The renderer's implementation for handling these will be a simple for-loop with a switch statement on the event type.